test/Shared/UserManagerTestBase.cs

```diff
@@ -1096,34 +1091,36 @@ public Task<IdentityResult> ValidateAsync(UserManager<TUser> manager, TUser user
             {
                 IdentityResultAssert.IsSuccess(await manager.CreateAsync(r));
             }
-            Assert.Equal(roles.Count, manager.Roles.Count(r => r.Name.StartsWith("CanQuerableRolesTest")));
-            var r1 = manager.Roles.FirstOrDefault(r => r.Name.StartsWith("CanQuerableRolesTest1"));
+            Assert.Equal(roles.Count, manager.Roles.Count(r => manager.GetRoleNameAsync(r).Result.StartsWith("CanQuerableRolesTest")));
+            var r1 = manager.Roles.FirstOrDefault(r => manager.GetRoleNameAsync(r).Result.StartsWith("CanQuerableRolesTest1"));
```

Comment:
Same comment as before about .Result. We should re-think how we test `IQueryables`.